### PR TITLE
Add Redis compability test runs

### PR DESCRIPTION
**Pull request by bjosv**
Run tests towards Redis version:
6.2-rc1, 6.0.9, 5.0.10, 4.0.14 and 3.2.7

### CODE REVIEW

File: test/eredis_tcp_SUITE.erl [hunk 1]
```diff
@@ -262,7 +262,8 @@ t_faulty_logical_database(Config) when is_list(Config) ->
     process_flag(trap_exit, true),
     Res = eredis:start_link("127.0.0.1", ?PORT, [{database, 99999999},
                                                  {reconnect_sleep, no_reconnect}]),
-    ?assertMatch({error, {select_error, {unexpected_response, <<"-ERR DB", _/binary>>}}}, Res),
+    %% Note: The error message content has changed from Redis 3 to Redis 4
```
Comment (zuiderkwast) [inline test/eredis_tcp_SUITE.erl:265]: .... so this test hasn't passed for a long time? Redis 4 must have been years ago...

**Review by zuiderkwast — COMMENTED** [review]
Looks good!

Do you think it's better to have this as a separate job than to combine it with the CI job?

**Comment by bjosv** [comment]
The problem is with the matrix handling in Github Actions. I didnt want a matrix of different erlang versions running a matrix of tests towards different redis versions, not sure if that gives anything more and it also seems a bit hard to filter out unwanted combinations. It felt more complex then having two different jobs.
An alternative could be to have both jobs in the same CI workflow/same file, but since they are quite large it felt easier to get an overview by separate them in files.
Each month there is 2000 min of free GithubAction time to use, quite a lot, but it would be interesting to see how much we have used sofar..

**Comment by zuiderkwast** [comment]
I understand. An option would be to include only certain (not all) combinations of OTP and Redis version in one matrix, but if you think it's better to have a separate job for Redis versions, go for it!

**Pull request merged**